common/guild_base.cpp

```diff
@@ -39,347 +88,310 @@ BaseGuildManager::~BaseGuildManager() {
+	info->tribute.id_2			 = 0xffffffff;
+	info->tribute.id_1_tier		 = 0;
+	info->tribute.id_2_tier		 = 0;
+	info->tribute.enabled		 = 0;
+	info->tribute.time_remaining = RuleI(Guild, TributeTime);
 
-	//clear out old `guilds` entry
-	auto results = m_db->QueryDatabase(query);
+	m_guilds[guild_id] = info;
 
-	//clear out old `guild_ranks` entries
-	query = StringFormat("DELETE FROM guild_ranks WHERE guild_id=%lu", (unsigned long)guild_id);
-	results = m_db->QueryDatabase(query);
+	return info;
+}
 
-	//escape our strings.
-	auto name_esc = new char[info->name.length() * 2 + 1];
-	auto motd_esc = new char[info->motd.length() * 2 + 1];
-	auto motd_set_esc = new char[info->motd_setter.length() * 2 + 1];
-	m_db->DoEscapeString(name_esc, info->name.c_str(), info->name.length());
-	m_db->DoEscapeString(motd_esc, info->motd.c_str(), info->motd.length());
-	m_db->DoEscapeString(motd_set_esc, info->motd_setter.c_str(), info->motd_setter.length());
-
-	//insert the new `guilds` entry
-	query = StringFormat("INSERT INTO guilds (id,name,leader,minstatus,motd,motd_setter) VALUES(%lu,'%s',%lu,%d,'%s', '%s')",
-		(unsigned long)guild_id, name_esc, (unsigned long)info->leader_char_id, info->minstatus, motd_esc, motd_set_esc);
-	results = m_db->QueryDatabase(query);
+bool BaseGuildManager::_StoreGuildDB(uint32 guild_id) 
+{
+	if(m_db == nullptr) {
+		LogError("Requested to store guild [{}] however there is no database object", guild_id);
+		return false;
+	}
 
-	if (!results.Success())
-	{
-		safe_delete_array(name_esc);
-		safe_delete_array(motd_esc);
-		safe_delete_array(motd_set_esc);
+	auto in = GetGuildByGuildID(guild_id);
+	if (!in) {
+		LogGuilds("Request to save guild id [{}] to the database however guild could not be found.", guild_id);
 		return false;
 	}
-	safe_delete_array(name_esc);
-	safe_delete_array(motd_esc);
-	safe_delete_array(motd_set_esc);
-
-	//now insert the new ranks
-	uint8 rank;
-	for(rank = 0; rank <= GUILD_MAX_RANK; rank++) {
-		const RankInfo &rankInfo = info->ranks[rank];
-
-		auto title_esc = new char[rankInfo.name.length() * 2 + 1];
-		m_db->DoEscapeString(title_esc, rankInfo.name.c_str(), rankInfo.name.length());
-
-        query = StringFormat("INSERT INTO guild_ranks "
-        "(guild_id,`rank`,title,can_hear,can_speak,can_invite,can_remove,can_promote,can_demote,can_motd,can_warpeace)"
-		" VALUES(%d,%d,'%s',%d,%d,%d,%d,%d,%d,%d,%d)",
-			guild_id, rank, title_esc,
-			rankInfo.permissions[GUILD_HEAR],
-			rankInfo.permissions[GUILD_SPEAK],
-			rankInfo.permissions[GUILD_INVITE],
-			rankInfo.permissions[GUILD_REMOVE],
-			rankInfo.permissions[GUILD_PROMOTE],
-			rankInfo.permissions[GUILD_DEMOTE],
-			rankInfo.permissions[GUILD_MOTD],
-			rankInfo.permissions[GUILD_WARPEACE]);
-		results = m_db->QueryDatabase(query);
-
-		if (!results.Success())
-		{
-			safe_delete_array(title_esc);
+
+	{
+		BaseGuildsRepository::Guilds out;
+		out.id = guild_id;
+		out.favor = in->tribute.favor;
+		GOUT(channel);
```

Comment:
It looks like this is a common pattern copying `GuildInfo*` to `BaseGuildsRepository::Guilds`. Maybe a private `BaseGuildManager` function to return a `BaseGuildsRepository::Guilds` from a  `GuildInfo*` argument would be better to avoid duplicating the logic everywhere this is done. Then the `GOUT` macro could also be removed since the assignments would only occur in one place.

e.g.
`BaseGuildsRepository::Guilds out = CreateGuildsRepoEntry(guild_id, in);`
(maybe with a more clever function name)

Reply:
Thanks for the discussion.  Made a function to look after this and found an oversight on my part in the process.  More evidence to have a single entity to reduce human error.